src/write.c

```diff
@@ -248,7 +248,7 @@ typedef struct avifEncoderData
 static void avifEncoderDataDestroy(avifEncoderData * data);
 
 // Returns NULL if a memory allocation failed.
-static avifEncoderData * avifEncoderDataCreate()
+static avifEncoderData * avifEncoderDataCreate(void)
```

Comment:
Some of these warning fixes are good, such as adding `void` for an empty parameter list, and fixing unintended "const casts" (e.g., at line 1005 below).